lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -762,6 +782,11 @@ sub cluster_jobs {
             }
             $p->cluster_jobs(jobs => $jobs);
         }
+        elsif ($pd->dependency eq OpenQA::JobDependencies::Constants::DIRECTLY_CHAINED) {
```

Comment:
I am not very clear with this logic. I just think if put the code `elsif ($pd->dependency eq OpenQA::JobDependencies::Constants::DIRECTLY_CHAINED)` behind of `if ($pd->dependency eq OpenQA::JobDependencies::Constants::CHAINED)` maybe looks better. I just say, not request you to modify :)